lib/transformers/i18n.js

```diff
@@ -0,0 +1,61 @@
+const JSON5 = require('json5')
+const yaml = require('js-yaml')
+const fs = require('fs')
+const path = require('path')
+
+const VUE_I18N_OPTION = '__i18n'
+
+function convert(source, lang) {
+  switch (lang) {
+    case 'yaml':
+    case 'yml':
+      return JSON.stringify(yaml.safeLoad(source), undefined, '\t')
+    case 'json5':
+      return JSON.stringify(JSON5.parse(source))
+    default:
+      return source
+  }
+}
+
+// Because the vue-i18n docs do not require a language
+// we have to try to parse the content as json first, then fall back to yaml
+function parseLanguageAndContent(block, filename) {
+  const langKnown = block.attrs && block.attrs.lang
+  let content = block.content
+
+  if (block.attrs && block.attrs.src) {
+    content = fs.readFileSync(path.resolve(filename, '../', block.attrs.src))
+  }
+
+  if (langKnown) {
+    return JSON.stringify(JSON.parse(convert(content, langKnown)))
+  }
+
+  try {
+    return JSON.stringify(JSON.parse(convert(content, 'json')))
+  } catch (_) {
+    return JSON.stringify(JSON.parse(convert(content, 'yaml')))
+  }
+}
+
+module.exports = {
+  process(blocks, vueOptionsNamespace, filename) {
+    const base = `${vueOptionsNamespace}.${VUE_I18N_OPTION} = []`
+    const codes = blocks.map(block => {
+      if (block.type === 'i18n') {
+        const value = parseLanguageAndContent(block, filename)
+          .replace(/\u2028/g, '\\u2028')
```

Comment:
What do these codes represent?

Reply:
They're line seperators, but really, I have no idea, I copied them from vue-i18n-jest

Reply:
ok, vue-i18n is battle tested, going to assume kazupon knows what he is doing